src/index.js

```diff
@@ -61,8 +62,9 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // `noRouterUpdate` flag can be set to avoid updating altogether,
     // which is useful for things like loading snapshots or very special
     // edge cases.
-    if(routing.path !== locationToString(window.location) &&
+    if(lastRoute !== routing.path && routing.path !== locationToString(window.location) &&
```

Comment:
Can you add a comment about why this is necessary? This approach generally looks fine to me (although I still don't understand why you need to pass `true` in the `updatePath` action above)